terra/config.py

```diff
@@ -18,39 +18,50 @@
 along with this program.  If not, see <http://www.gnu.org/licenses/>
 
 """
-from gi.repository import Gdk
+
 from base64 import b64encode, b64decode
 import ConfigParser
-from defaults import ConfigDefaults
 import os
 
-__terra_data_directory__ = '/usr/local/share/terra/'
-__terra_app_directory__ = '/usr/local/share/applications/'
-__version__ = '0.2.0'
+from gi.repository import Gdk
 
-class ConfigManager():
-    config_dir_path = os.path.join(os.environ['HOME'], '.config', 'terra')
-    config_file_path = os.path.join(config_dir_path, 'main.cfg')
+from terra import (__version__)
+from terra.defaults import ConfigDefaults
+from terra.handler import TerraHandler
+
+
+class ConfigManager:
+    version = __version__
+    config_file_name = 'main.cfg'
 
-    config = ConfigDefaults
-    config.read(config_file_path)
     defaults = ConfigDefaults
+    config = ConfigDefaults
 
     callback_list = []
     use_fake_transparency = False
     disable_losefocus_temporary = False
 
-    data_dir = __terra_data_directory__
-    version = __version__
+    @classmethod
+    def __init__(cls):
+        # Load the user config, if any.
+        config_path = TerraHandler.get_config_path()
+        if os.path.exists(config_path):
+            config_file_path = os.path.join(config_path, cls.config_file_name)
+            if os.path.exists(config_file_path):
+                cls.config.read(config_file_path)
 
-    @staticmethod
-    def init():
-        ConfigParser.SafeConfigParser({})
-        ConfigManager.save_config()
+        # TODO: Remove once the code no longer requires the config files!
+        # Now save the config file.
+        cls.save_config()
 
-    @staticmethod
-    def get_sections():
-        return (ConfigManager.config.sections())
+    @classmethod
+    def get_sections(cls):
+        # TODO: Remove! For some reason, on my setup the `sections()` method
+        # raises a "tuple assignment index out of range" exception.
+        # Reading items from the config seems to solve the issue.
```

Comment:
I solved it on the new manager I created. Maybe I'll just commit it as `settings.py`.  The new one seems to work just fine, 

I replace the whole thing with this:
```
    @property
    def sections(self):
        """
        :return dict:
        """
        sections_list = self._defaults.keys()
        for section in self.iterkeys():
            if section not in sections_list:
                # Delete default values.
                if sections_list[section] is None:
                    del sections_list[section]
                    continue

                sections_list.append(section)

        return sections_list
```
Where that "Delete default values" part simply allows one to remove a whole section from the default settings, for example a layouts section. But it should not really be required to have such a thing, maybe I'll just remove it.